### PR TITLE
TINY-2236: Added new `listbox` dialog component and added back the ability to use nested link_list items

### LINKED ISSUE #4822: BUG: Link list sub-menus in ver. 5
Since TinyMCE 4.0.27 it is possible to have sub-menus within the link list. 

From TinyMCE 5.0 sub-menus aren't visible.

All OS and browsers are affected.

Try http://fiddle.tinymce.com/wleaab/102 (from originale fiddle as suggested in documentation at https://www.tiny.cloud/docs/plugins/link/) or http://fiddle.tinymce.com/WBgaab (sample with same configuration suggested in documentation).

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-2236

Description of Changes:
This adds a new `listbox` UI component that uses the TinyMCE menu components to generate a `selectbox` style dropdown. It then updates all the relevant plugins (`image`, `link` and `table`) to use `listbox` components instead of `selectbox`. This means that only `codesample` and `template` plugins (plus premium plugins like `a11ychecker`) should be using the `selectbox` component now, as it didn't make sense to switch them over.

Here's an example of what it looks like:
image

**Note:** Ideally we should focus the selected option, however the menu components don't allow for the "focused" item to be set. As such we're going to log that as an improvement to add in a later release.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):
Fixes #4822

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/ui/button/MenuButton.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ const renderMenuButton = (spec: MenuButtonSpec, prefix: string, backstage: UiFac
   tooltip: spec.tooltip,
   // https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-2/menubar-2.html
   role,
-  fetch: (callback) => {
+  fetch: (comp, callback) => {
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/ui/button/MenuButton.ts:45]: Was wondering if `comp` should have an underscore to be clear it isn't used?

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/changelog.txt [hunk 2]
```diff
@@ -10,6 +10,8 @@ Version 5.5.0 (TBD)
     Added the `origin` property to the `ObjectResized` and `ObjectResizeStart` events, to specify which handle the resize was preformed on #TINY-6242
     Added new StyleSheetLoader `unload` and `unloadAll` APIs to allow loaded stylesheets to be removed #TINY-3926
     Added a new `contextmenu_avoid_overlap` setting to allow contextmenus to avoid overlapping matched nodes #TINY-6036
+    Added new `listbox` dialog UI component for rendering a `selectbox` like list that accepts nested items #TINY-2236
+    Added back the ability to use nested items in `image_class_list`, `link_class_list`, `link_list`, `table_class_list`, `table_cell_class_list`, and `table_row_class_list` settings #TINY-2236
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/changelog.txt:14]: Yeah, it should, I just missed `the` sorry. Thanks for the catch!

**Review by metricjs — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/button/MenuButton.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/themes/silver/main/ts/ui/button/MenuButton.ts]: Fixed

File: modules/tinymce/changelog.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/changelog.txt]: Fixed

**Pull request merged**